### PR TITLE
feat(AUTH-CPC-916): give feedback while user is being created

### PR DESCRIPTION
**Pull request by RichardDanon**
JIRA: [link to jira ticket](https://champlainsaintlambert.atlassian.net/browse/CPC-916?atlOrigin=eyJpIjoiNDVjNWFjNWNlOTViNGJjZGFmODUxNTRhNTAxZTZlMzUiLCJwIjoiaiJ9)

## Context:

This ticket was made because the feedback that the user was getting while creating their account was very minimal. It did not properly guide the users through the process.  

## Changes
What are the various changes and what other modules do those changes effect.

1. Pop up when submission is successful
3. Change the Error display for the password
5. Redirect the verification page generated from the email to the log-in page
7. Test email verification and user verification process
9. The password now shows while filling out the form
10. Added password Strength tracker in signup and password reset

.## Before and After UI (Required for UI-impacting PRs)

Before:
image
image
image

After:
image
image
image
image
image
image
image
image

### CODE REVIEW

File: auth-service/src/test/java/com/petclinic/authservice/businesslayer/AuthServiceUserServiceTests.java [hunk 1]
```diff
@@ -202,7 +217,57 @@ public void testUpdateResetPasswordToken_UserNotFound() {
...
+            userService.updatePassword(newPassword, token);
+        });
+
+        assertEquals("Could not find any customer with the token " + token, exception.getMessage());
+
+        verify(userRepo).findById(validToken.getUserIdentifier());
+    }
+
 
+// Next Story
```
Comment (cubajosh) [inline auth-service/src/test/java/com/petclinic/authservice/businesslayer/AuthServiceUserServiceTests.java:246]: What does this next story part mean?

**Review by cubajosh — DISMISSED** [review]
Everything looks awesome, good job man.

File: api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/AuthServiceClient.java [hunk 2]
```diff
@@ -121,16 +121,27 @@ public Mono<OwnerResponseDTO> createUser (Mono<Register> model) {
...
                 .get()
                 .uri(authServiceUrl + "/users/verification/{token}", token)
                 .retrieve()
-                .onStatus(HttpStatusCode::is4xxClientError,
-                        n -> rethrower.rethrow(n,
+                .onStatus(
+                        HttpStatusCode::is4xxClientError,
+                        n -> rethrower.rethrow(
+                                n,
                                 x -> new GenericHttpException(x.get("message").toString(), BAD_REQUEST))
```
Comment (DylanBrass) [inline api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/AuthServiceClient.java:134]: Why did you put BAD_REQUEST, iifc you can but n.getStatusCode ?

File: api-gateway/src/main/resources/static/scripts/auth/reset-pwd-form/reset-pwd-form.template.html [hunk 3]
```diff
@@ -24,12 +28,78 @@ <h3 class="text-danger text-center" ng-repeat="error in errorMessages">{{ error
...
+
+    passwordStrength.textContent = strengthText;
+    passwordStrength.className = `password-strength strength-${strength}`;
+  }
+
+  function calculatePasswordStrength(password) {
+
+    const pattern = /^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%^&+=*!()_.<>,{}])(?=\S+$).{8,}$/;
+
+    if (pattern.test(password)) {
```
Comment (DylanBrass) [inline api-gateway/src/main/resources/static/scripts/auth/reset-pwd-form/reset-pwd-form.template.html:60]: Can you explain what .test does ?

File: api-gateway/src/main/resources/static/scripts/auth/signup/signup-form.template.html [hunk 4]
```diff
@@ -2,68 +2,177 @@
...
 <!--        * Date: 22/09/21-->
 <!--        * Ticket: feat(AUTH-CPC-102)_signup_user-->
-<div style="display: flex; justify-content: center">
-    <div>
+<div style="display: flex; justify-content: center;">
+    <div style="width: 400px;">
         <h2 class="text-center">Signup</h2>
-        <h3 class="text-danger text-center" ng-repeat="error in errorMessages">{{ error }}</h3>
-        <div class="d-flex justify-content-center">
-            <form class="form-horizontal" ng-submit="$ctrl.add()" style="max-width: 25em; justify-content: center;">
```
Comment (DylanBrass) [inline api-gateway/src/main/resources/static/scripts/auth/signup/signup-form.template.html:10]: Why did you remove all of this ?

**Review by DylanBrass — APPROVED** [review]
Good change, and nice work left a few questions and comments.

File: auth-service/src/test/java/com/petclinic/authservice/businesslayer/AuthServiceUserServiceTests.java (hunk 1 above)
Comment (RichardDanon) [reply auth-service/src/test/java/com/petclinic/authservice/businesslayer/AuthServiceUserServiceTests.java:246]: it means its a test that works but since some of the exceptions being used are wrong I'll do it in the next story concerning it.

File: api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/AuthServiceClient.java (hunk 2 above)
Comment (RichardDanon) [reply api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/AuthServiceClient.java:134]: It's a method used on a regular expression object to check if the string I'm giving it matches the regular expression pattern I am specifying. In this case, it is checking to see if the password matches the regex.

File: api-gateway/src/main/resources/static/scripts/auth/reset-pwd-form/reset-pwd-form.template.html (hunk 3 above)
Comment (RichardDanon) [reply api-gateway/src/main/resources/static/scripts/auth/reset-pwd-form/reset-pwd-form.template.html:60]: It's a method used on a regular expression object to check if the string I'm giving it matches the regular expression pattern I am specifying. In this case, it is checking to see if the password matches the regex.

File: api-gateway/src/main/resources/static/scripts/auth/signup/signup-form.template.html (hunk 4 above)
Comment (RichardDanon) [reply api-gateway/src/main/resources/static/scripts/auth/signup/signup-form.template.html:10]: I didn't I just modified it a bit and moved it around except the <h3>

**Review by DylanBrass — APPROVED** [review]

File: api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/AuthServiceClient.java (hunk 2 above)
Comment (RichardDanon) [reply api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/AuthServiceClient.java]: Done

**Review by cubajosh — APPROVED** [review]

**Review by AmirHamzahMirza — APPROVED** [review]
Minor changes by the author, not to be commented on
